components/StorageFileList.vue

```diff
@@ -1,27 +1,60 @@
 <script setup lang="ts">
 import { FileObject } from "@supabase/storage-js";
+import { FilterParams } from "@/types/FilterParams";
+import { QueryParams } from "@/types/QueryParams";
 
 const sortColumn = useSortColumn();
 const sortOrder = useSortOrder();
 const layoutType = useLayoutType();
 const selectedFiles = useSelectedFiles();
 
+const filters = ref<FilterParams>({
+  name: null,
+  types: null,
+  sizeMin: null,
+  sizeMax: null,
+  dates: null,
+});
+
 const computedClass = computed(() => {
   return layoutType.value === "grid"
     ? "files__list--grid"
     : "files__list--list";
 });
 
+const queryParameters = computed(() => {
+  const queryObject: QueryParams = {
+    key: sortColumn.value,
+    order: sortOrder.value,
+  };
+  if (filters.value.name) {
```

Comment:
what happen if user clear only `name` by removing it from the input? is it reseted somewhere else?

Reply:
Not sure if I understand the question. User has to click Confirm if he enters some filters or changes it. Clicking Confirm and Clear filters buttons emit the values to filters ref in StorageFileList, then queryParams as computed gets updated and then call to server/api/files is made, so list gets updated.

Now I see the UX might be not the best cuz when user enters one filter, eg name, then comes back to filters and deletes it - so that there is no active filter - the Confirm button is disabled and he has to click Clear filters.

Reply:
I haven't check before but right, the value of, for example, `name` is updated in `queryParameters` even if user type back the name and `filters.value.name === ''` 🤔 but I don't know how it happens because `queryParameters.name` should be updated only if `filters.value.name` is truthy 😄 I have to missing something while looking at it 🤷 

and I agree with your UX suggestion - I had this moment of inconvenience when I cleared filters manually and wasn't able to save them until found `clear` button 😄

Reply:
good point with that truthy values 👍 
I updated the UX so that the confirm is disabled only when dates are not correct